### ISSUE TITLE
nuxt build no throw error when process.env.CI = true

### ISSUE THREAD

**Issue by echoulen** [nuxt/nuxt.js#5623]
### Version

[v2.6.3](https://github.com/nuxt.js/releases/tag/v2.6.3)

### Reproduction link

[https://github.com/nuxt/nuxt.js/blob/dev/packages/webpack/src/builder.js#L150](https://github.com/nuxt/nuxt.js/blob/dev/packages/webpack/src/builder.js#L150)

### Steps to reproduce

- In Travis CI, the `process.env.ci` is `true`
- nuxt build, expect error happened, but build complete without error

### What is expected ?

throw the error

### What is actually happening?

return the `Promise.reject` but no catch

### Additional comments?

For now, the workaround is using the module https://github.com/echoulen/nuxt-build-error-catch-plugin to catch the `Promise.reject`

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c9131">#c9131</a>)</em></sub></div>

**Comment by lorado** [comment]
Oh just got the same issue.

I am using GitLab CI, didn't even notices, but it seems that GitLab CI automatically defines this CI environment variable. As result my build exited without error, but produced no build. I was wondering, what is happening, because I didn't noticed error on my local machine... 

This kinda not good... I expect that CI Pipeline fails, if build is not generated properly. Please fix it

**Comment by leipert** [comment]
We just ran into the same issue. I solved it by setting [`build.quiet`](https://nuxtjs.org/api/configuration-build/#quiet) to `false`.

Here you can see an example of the build being green, without any error, while our master branch was currently broken at the time: https://gitlab.com/gitlab-org/design.gitlab.com/-/jobs/252288505

![Screenshot 2019-07-16 at 11 04 15](https://user-images.githubusercontent.com/2906107/61281167-7c6fc980-a7b9-11e9-879a-09ba6e86d38a.png)

Here is the commit, where I turned on `build.quiet: false`: https://gitlab.com/gitlab-org/design.gitlab.com/-/jobs/252363564

![Screenshot 2019-07-16 at 11 07 16](https://user-images.githubusercontent.com/2906107/61281423-e7210500-a7b9-11e9-94dd-f338c4d8bb9e.png)

So in summary, our SCSS compilation failed and it didn't lead to build errors, because `build.quiet` is per default true on CI which leads to this error.

1. I am wondering why the default is true, who doesn't want build output during the CI job?
2. Also the problem: Why does it swallow the error?

- - -

My tone might sound a bit harsh, not intended, just didn't have my coffee yet 😆 Thank you for the great work on nuxt!

**Comment by clarkdo** [comment]
I’ll have a look shortly
